system/jlib/jevent.cpp

```diff
@@ -146,13 +150,68 @@ static constexpr EventAttrInformation attrInformation[] = {
 
 static_assert(_elements_in(attrInformation) == EvAttrMax);
 
+EventContext queryEventContext(EventType event)
+{
+    switch (event)
+    {
+    case EventDaliChangeMode:
+    case EventDaliCommit:
+    case EventDaliConnect:
+    case EventDaliEnsureLocal:
+    case EventDaliGet:
+    case EventDaliGetChildren:
+    case EventDaliGetChildrenFor:
+    case EventDaliGetElements:
+    case EventDaliSubscribe:
+        return EventCtxDali;
+    case EventIndexLookup:
+    case EventIndexLoad:
+    case EventIndexEviction:
+    case MetaFileInformation:
+        return EventCtxIndex;
```

Comment:
I would add a field to the eventInformation and use that to implement - then the information is all in one place.